### ISSUE TITLE
Reduce the verbosity of unit tests

### ISSUE THREAD

**Issue by DiegoEliasCosta** [DiegoEliasCosta/spotjmhbugs#4]
After migrating to SpotBugs 3.1.12 unit-tests now print the method byte-code.
This should be configured to be printed only when running in DEBUG mode with build systems.

Example of 
```
de.heidelberg.pvs.diego.jmh.JMHSample_08_DeadCode.<init> : ()V
  0 aload_0         []::[< Lde/heidelberg/pvs/diego/jmh/JMHSample_08_DeadCode;, IP, r0 >]
  1 invokespecial   [< Lde/heidelberg/pvs/diego/jmh/JMHSample_08_DeadCode;, IP, r0 >]::[< Lde/heidelberg/pvs/diego/jmh/JMHSample_08_DeadCode;, IP, r0 >]
  4 aload_0         []::[< Lde/heidelberg/pvs/diego/jmh/JMHSample_08_DeadCode;, IP, r0 >]
  5 ldc2_w          [< Lde/heidelberg/pvs/diego/jmh/JMHSample_08_DeadCode;, IP, r0 >]::[< Lde/heidelberg/pvs/diego/jmh/JMHSample_08_DeadCode;, IP, r0 >]
  8 putfield        [< Lde/heidelberg/pvs/diego/jmh/JMHSample_08_DeadCode;, IP, r0 >, < D, 3.141592653589793 >]::[< Lde/heidelberg/pvs/diego/jmh/JMHSample_08_DeadCode;, IP, r0 >]
 11 return          []::[< Lde/heidelberg/pvs/diego/jmh/JMHSample_08_DeadCode;, IP, r0 >]
```

**Comment by isopov** [comment]
This verbosity comes with bumping spotbugs version from 3.1.1 to 3.1.2

**Comment by isopov** [comment]
https://github.com/spotbugs/spotbugs/compare/3.1.1...spotbugs:3.1.2 - link to compare of these two versions (24 commits in 3.1.2)

**Comment by isopov** [comment]
I suspect the following commit - https://github.com/spotbugs/spotbugs/commit/88fb657b2055d062d09b3952310f58356304619e

**Comment by DiegoEliasCosta** [comment]
Yeah... that puzzles me a bit. The problem is that apparently `SpotBugsRule` is the only available entry-point for unit-tests in SpotBugs and there is no way to customize the detectors that are running in the tests. 

I found some detectors that have unguarded `print` statements that will be printed every time we run some unit tests.
- [TestingGround](https://github.com/spotbugs/spotbugs/blob/3883a7b750fb339577be073bc45e36b6f268777b/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/TestingGround.java)
- [CheckCalls](https://github.com/spotbugs/spotbugs/blob/07bf864b83083c467e29f1b2de58a2cf5aa5c0d6/spotbugs/src/main/java/edu/umd/cs/findbugs/detect/CheckCalls.java)

The easiest solution would be to send a PR to SpotBugs guarding the print statements with a `DEBUG` variable, which is how most detectors are implemented. Another option would be to implement better control over the detectors that are running, but that can take quite some time.

I will send them a PR tomorrow.
